Build failure on `wasm32-unknown-emscripten` due to overlapping `cfg` conditions in `global_style_data.rs`

**File:** `global_style_data.rs`

## Summary
Building Stylo on the target **`wasm32-unknown-emscripten`** fails because both `#[cfg(unix)]` and `#[cfg(all(target_arch = "wasm32", not(feature = "gecko")))]` evaluate to **true** on this target. This causes:

- Duplicate definitions of `PlatformThreadHandle` (`E0428`)
- Type mismatch when pushing thread handles (`E0308`)

The root issue is that `wasm32-unknown-emscripten` is treated as **`unix`** by Rust, so the unix thread-handle path and the wasm thread-handle path overlap.

## Reproduction
Target:
```
wasm32-unknown-emscripten
```
Error excerpt:
```
error[E0428]: the name `PlatformThreadHandle` is defined multiple times
error[E0308]: mismatched types: expected `RawPthread`-like, found `DummyThreadHandle`
```

Full error:
```
error[E0428]: the name `PlatformThreadHandle` is defined multiple times
  --> /Users/softmarshmallow/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/stylo-0.9.0/global_style_data.rs:41:1
   |
23 | pub type PlatformThreadHandle = RawPthread;
   | ------------------------------------------- previous definition of the type `PlatformThreadHandle` here
...
41 | pub type PlatformThreadHandle = DummyThreadHandle;
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ `PlatformThreadHandle` redefined here
   |
   = note: `PlatformThreadHandle` must be defined only once in the type namespace of this module

error[E0308]: mismatched types
   --> /Users/softmarshmallow/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/stylo-0.9.0/global_style_data.rs:155:26
    |
155 |             handles.push(handle);
    |                     ---- ^^^^^^ expected `u32`, found `DummyThreadHandle`
    |                     |
    |                     arguments to this method are incorrect
    |
```


## Root Cause
In `global_style_data.rs`:
```rust
#[cfg(unix)]
pub type PlatformThreadHandle = RawPthread;

#[cfg(all(target_arch = "wasm32", not(feature = "gecko")))]
pub type PlatformThreadHandle = DummyThreadHandle;
```
On `wasm32-unknown-emscripten`, both conditions are true → duplicate type alias.

Same overlap happens inside `get_thread_handles()` where two `let handle = ...` bindings are emitted.

## Proposed Fix
Restrict the unix path to exclude wasm:
```rust
#[cfg(all(unix, not(target_arch = "wasm32")))]
pub type PlatformThreadHandle = RawPthread;
```
And similarly inside `get_thread_handles()`:
```rust
#[cfg(all(unix, not(target_arch = "wasm32")))]
let handle = join_handle.as_pthread_t();
```

This ensures that on wasm targets, only the `DummyThreadHandle` path is active.

## Result
After applying this change, the crate compiles successfully under `wasm32-unknown-emscripten`.



## Context
- version: 0.9.0

--- a/style/global_style_data.rs
+++ b/style/global_style_data.rs
@@ -11,7 +11,7 @@ use crate::parallel::STYLE_THREAD_STACK_SIZE_KB;
 use crate::shared_lock::SharedRwLock;
 use crate::thread_state;
 use parking_lot::{Mutex, RwLock, RwLockReadGuard};
-#[cfg(unix)]
+#[cfg(all(unix, not(target_arch = "wasm32")))]
 use std::os::unix::thread::{JoinHandleExt, RawPthread};
 #[cfg(windows)]
 use std::os::windows::{io::AsRawHandle, prelude::RawHandle};
@@ -19,7 +19,7 @@ use std::{io, thread};
 use thin_vec::ThinVec;
 
 /// Platform-specific handle to a thread.
-#[cfg(unix)]
+#[cfg(all(unix, not(target_arch = "wasm32")))]
 pub type PlatformThreadHandle = RawPthread;
 /// Platform-specific handle to a thread.
 #[cfg(windows)]
@@ -142,7 +142,7 @@ impl StyleThreadPool {
         lazy_static::initialize(&STYLE_THREAD_POOL);
 
         for join_handle in STYLE_THREAD_JOIN_HANDLES.lock().iter() {
-            #[cfg(unix)]
+            #[cfg(all(unix, not(target_arch = "wasm32")))]
             let handle = join_handle.as_pthread_t();
             #[cfg(windows)]
             let handle = join_handle.as_raw_handle();

